crates/kotlin-ffi/src/lib.rs

```diff
@@ -68,6 +78,72 @@ uniffi::custom_type!(FFIBytes, String, {
     lower: |obj| obj.to_string(),
 });
 
+#[derive(Debug, uniffi::Record)]
+pub struct RouteUiFields {
+    pub route: Vec<TxnDetails>,
+    pub bridge: Vec<TransactionFee>,
+    pub initial: TxnDetails,
+    pub local_total: Amount,
+}
+
+impl From<yttrium::chain_abstraction::client::RouteUiFields> for RouteUiFields {
+    fn from(source: yttrium::chain_abstraction::client::RouteUiFields) -> Self {
+        Self {
+            route: source.route.into_iter().map(Into::into).collect(),
+            bridge: source.bridge,
+            initial: source.initial.into(),
+            local_total: source.local_total,
+        }
+    }
+}
+
+#[derive(Debug, uniffi::Record)]
+pub struct TxnDetails {
+    pub transaction: CATransaction,
+    pub estimate: Eip1559Estimation,
+    pub fee: TransactionFee,
+}
+
+impl From<yttrium::chain_abstraction::client::TxnDetails> for TxnDetails {
+    fn from(source: yttrium::chain_abstraction::client::TxnDetails) -> Self {
+        Self {
+            transaction: source.transaction,
+            estimate: source.estimate.into(),
+            fee: source.fee,
+        }
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, uniffi::Record)]
+pub struct Eip1559Estimation {
+    /// The base fee per gas.
+    pub max_fee_per_gas: FFIU128,
+    /// The max priority fee per gas.
+    pub max_priority_fee_per_gas: FFIU128,
+}
+
+impl From<alloy::providers::utils::Eip1559Estimation> for Eip1559Estimation {
+    fn from(source: alloy::providers::utils::Eip1559Estimation) -> Self {
+        Self {
+            max_fee_per_gas: FFIU128::from(source.max_fee_per_gas),
+            max_priority_fee_per_gas: FFIU128::from(
+                source.max_priority_fee_per_gas,
+            ),
+        }
+    }
+}
+
+// uniffi::custom_type!(Eip1559Estimation, FfiEip1559Estimation, {
+//     try_lift: |val| Ok(Eip1559Estimation {
+//         max_fee_per_gas: val.max_fee_per_gas.to(),
+//         max_priority_fee_per_gas: val.max_priority_fee_per_gas.to(),
+//     }),
+//     lower: |obj| FfiEip1559Estimation {
+//         max_fee_per_gas: U128::from(obj.max_fee_per_gas),
+//         max_priority_fee_per_gas: U128::from(obj.max_priority_fee_per_gas),
+//     },
+// });
```

Comment:
Delete this?

Reply:
Same thing; left as a placeholder in-case we want to use it